### PR TITLE
Applicant update

### PR DESCRIPTION
**Pull request by reginafcompton**
This PR adds functionality to the admin view, so that administrators can update the `owned_pin` and/or `street` of an applicant's owned address. 

lots

This PR includes a new UpdatedEntity model, which tracks changes, the admin who made them, and when it happened. 

***
I made use of an `inline_formset` for updating the applicant's `owned_pin`, since it made it easy to create a form that respects the relationship between the Applicant and the Address models. However, I could have done something similar be creating a simple ModelForm for the applicant and instantiating it with the Application instance found in the view. So.....I am not going to fight for one over the other, but I could see how the formset solution is somewhat unnecessary, since we'd never need to render multiple, identical forms.

### CODE REVIEW

File: lots_admin/admin.py [hunk 1]
```diff
@@ -1,17 +1,22 @@
...
-from lots_admin.models import Address, Application, Lot, ApplicationStep, Review, DenialReason, ApplicationStatus
+from lots_admin.models import Address, Application, Lot, ApplicationStep, Review, DenialReason, ApplicationStatus, UpdatedEntity
 
 class AdminApplication(admin.ModelAdmin):
     search_fields = ['first_name', 'last_name', 'id']
 
 class AdminApplicationStatus(admin.ModelAdmin):
     search_fields = ['application__first_name', 'application__last_name', 'application__id']
 
+class AdminUpdatedEntity(admin.ModelAdmin):
```
Comment (hancush) [inline lots_admin/admin.py]: trivia time! if it makes you twitch to read `UpdatedEntitys` in the admin, you can [set a meta option](https://docs.djangoproject.com/en/1.10/ref/models/options/#verbose-name-plural) to define the plural form, like:

```python
class UpdatedEntity(models.Model):
    ...
    class Meta:
        verbose_name_plural = 'UpdatedEntities'
```

Comment (hancush) [reply lots_admin/admin.py]: (this saved my sanity over in bga, hehe)

File: lots_admin/views.py [hunk 2]
```diff
@@ -911,12 +914,51 @@ def review_EDS(request, application_id):
...
+    ApplicantFormSet = inlineformset_factory(Address, 
+                                             Application, 
+                                             fk_name='owned_address', 
+                                             fields=('owned_pin',), 
+                                             extra=0, 
+                                             can_delete=False, 
+                                             widgets = {
+                                                'owned_pin': forms.TextInput(attrs={'class': 'form-control'}),
+                                             })
+    application_formset = ApplicantFormSet(instance=application.owned_address)
```
Comment (hancush) [inline lots_admin/views.py]: what happens here if more than one application has the same owned address?

File: lots_admin/forms.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        add_info = {
+            'street': street,
+            'street_number': street_number,
+            'street_dir': street_dir,
+            'street_name': street_name,
+            'street_type': street_type,
+        }
+
+        address = super(AddressUpdateForm, self).save(*args, **kwargs)
```
Comment (hancush) [inline lots_admin/forms.py]: if this address is in use by multiple applications, is it safe to update the address object for all those applications?

File: lots_admin/views.py [hunk 4]
```diff
@@ -911,12 +914,51 @@ def review_EDS(request, application_id):
...
+
+    if request.method == 'POST':
+        update_info = {'admin': request.user}
+        application_formset = ApplicantFormSet(request.POST, instance=application.owned_address)
+        if application_formset.is_valid():
```
Comment (hancush) [inline lots_admin/views.py]: is it ok that we save input from one form, if the other form is not valid?

File: lots_admin/admin.py (hunk 1 above)
Comment (reginafcompton) [reply lots_admin/admin.py]: oh, lovely.

File: lots_admin/views.py (hunk 4 above)
Comment (reginafcompton) [reply lots_admin/views.py]: ah, good point.

**Comment by reginafcompton** [comment]
I abandoned the inline formset (as we both suggest above!), mainly because this use case does not call for a formset, since the application (as determined by a related address) should have only one `owned_pin`.

File: lots_admin/forms.py (hunk 3 above)
Comment (reginafcompton) [reply lots_admin/forms.py]: Yes, multiple applicants should not share an address (that's cheating!), but if they do, then the Address entity should still be updated. It's still a bad address, even if it relates to many applications.

File: lots_admin/views.py (hunk 2 above)
Comment (reginafcompton) [reply lots_admin/views.py]: See comment below.

File: lots_admin/views.py [hunk 5]
```diff
@@ -32,11 +32,14 @@
 from django.core.mail import EmailMultiAlternatives
 from django.db.models import Q
 from django.core.exceptions import MultipleObjectsReturned
+from django.forms import inlineformset_factory
```
Comment (hancush) [inline lots_admin/views.py]: we can ditch this import!

**Pull request merged**